### ISSUE TITLE
Convert game to out-of-source build

### ISSUE THREAD

**Issue by jameskr97** [etternagame/etterna#503]
In Etterna's current state, the game **MUST** be built as an in-source build. While some may say this is not an in-source build since all the compilation obj files are placed into respective directories within the build directory, the output binary itself must be placed in relation to the root of the source tree. This output location allows the game my access the various asset directories like `Songs`, `Scripts`, `Themes`, `NoteSkins` and so on.

I am proposing, in order to fix this issue, that all directories which contain game assets/resources into a separate directory named `Assets` or `Resources` in order for all game assets to be in the same location, and to be away from the root of the source tree. Ideally, once cmake generation and compilation have been completed, every component which is required to have the game run will be placed inside the build directory.

**Comment by Kangaroux** [comment]
Somewhat of a separate issue but I think there should be two folders, one for game assets that may be overwritten on installs, and another for user assets. This draws a clear line for files that can and cannot be edited. This would also mean merging the user and game assets together. For example, the game assets folder would have the default noteskins, but the user folder could have its own noteskins folder as well. At startup, noteskins from both the game and user folders would be loaded. Same thing applies to themes. Songs and save data would exist in the user assets folder.

**Comment by jameskr97** [comment]
@Kangaroux I agree with this. I think back to Minecraft. They have their core assets that are hidden from user view, and always available, then a separate area for where use customized assets may be placed.

**Comment by nico-abram** [comment]
> Ideally, once cmake generation and compilation have been completed, every component which is required to have the game run will be placed inside the build directory.

I think this was mentioned when we talked about this in discord at some point, but I just wanted to leave it in writing here:

This sounds somewhat error-prone, since for example theme assets like images or lua scripts will be copied and the game will run using a copy. Therefore, to test changes ingame we would have to modify the build directory copy, but to actually commit changes we would have to copy them back to their original folders, which sounds like an error-prone process. Pherhaps what we could do is add a flag or hack to make it so when the game runs it walks parent directories until it finds one that has child directories it expects like /Data or /Themes , and then look for all the assets it needs from there.
